tests/test_date_utils.py

```diff
@@ -17,16 +17,25 @@
 # along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
 
+"""Tests for DoJSON utilities."""
 
-from inspirehep.dojson.utils import create_valid_date
+from inspirehep.utils.date import create_valid_date, create_earliest_date
 
 
 def test_create_valid_date():
     """Test date validator used in dojson conversions."""
-    assert create_valid_date(1877) == '1877-01-01'
-    assert create_valid_date("1877") == '1877-01-01'
-    assert create_valid_date("1877-02") == '1877-02-01'
-    assert create_valid_date("1877-02-03") == '1877-02-03'
-    assert create_valid_date(1977) == '1977-01-01'
-    assert create_valid_date("1977") == '1977-01-01'
-    assert create_valid_date("1977-06-02") == '1977-06-02'
+    assert create_valid_date(1877) == '1877'
+    assert create_valid_date('1877') == '1877'
+    assert create_valid_date('1877-02') == '1877-02'
+    assert create_valid_date('1877-02-03') == '1877-02-03'
+    assert create_valid_date(1977) == '1977'
+    assert create_valid_date('1977') == '1977'
+    assert create_valid_date('1977-06-02') == '1977-06-02'
+    assert create_valid_date('1977-06-022') == '1977-06'
+    assert create_valid_date('1977-06-220') == '1977-06'
+
+
+def test_create_earliest_date():
+    """Test the date validator that excepts list."""
+    assert create_earliest_date([1877, '2002-01-05']) == '1877'
+    assert create_earliest_date(['1877-02-03', '1877']) == '1877-02-03'
```

Comment:
To answer you question, so yeah, we could return `01`, but as you said that it would perhaps be a bad assumption to make, perhaps it is better to not add it?